finger-control/fingercontrol.py

```diff
@@ -3,89 +3,91 @@
 import mss
 import numpy as np
 import os
-import picamera
-import picamera.array
 import pyautogui
 import sys
 import time
+from camera import CameraFactory
 
 # enable debug output
-debug=False
-debugSwitches = ['-v', 'debug', '--verbose', 'DEBUG']
-
-# Camera resolution
-res_width=1280
-res_height=720
+debug = False
+DEBUG_SWITCHES = ['-v', 'debug', '--verbose', 'DEBUG']
 
 # Screen resolution
-screen_width=1280
-scree_height=720
+SCREEN_WIDTH=1280
+SCREEN_HEIGHT=720
 
 # Definition of area of interest
 # This is set to the area of the projection
-x_off=280
-y_off=0
-width=680
-height=340
+X_OFFSET=280
+Y_OFFSET=0
+WIDTH=680
+HEIGHT=340
 
-# Treshholds
-reflections_tresh=30
-movement_tresh=100000
+# Thresholds
+REFLECTIONS_THRESHOLD=30
+MOVEMENT_THRESH=100000
 
-# Image processing iteration
+# Image processing iteration (for debugging only)
 iteration = 0
 
-def init_camera(camera):
-    camera.resolution = (res_width, res_height)
-    time.sleep(3)
 
-def take_picture(camera):
-    cap = picamera.array.PiRGBArray(camera)
-    camera.capture(cap, format="bgr")
+def to_grayscale(image):
+    result = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+    return result
+
+
+def clip_projection_area(image):
+    result = image[Y_OFFSET : (Y_OFFSET + HEIGHT), X_OFFSET : (X_OFFSET + WIDTH)]
+    return result
+
+
+def preprocess_image(image):
+    clipped = clip_projection_area(image)
```

Comment:
Nur Bilder von der Camera müssen zugeschnitten werden, oder?

Eventuell muss man das Schneiden direkt in der Kamera machen.